docs(uk): translate useEffect.md to Ukrainian

Andrii256 opened the pull request:
Translate useEffect section to Ukrainian language

Andrii256 committed:
Translate useEffect page

alinkedd commented on src/content/reference/react/useEffect.md:
```diff
@@ -40,45 +40,47 @@ function ChatRoom({ roomId }) {
-* `setup`: The function with your Effect's logic. Your setup function may also optionally return a *cleanup* function. When your component is added to the DOM, React will run your setup function. After every re-render with changed dependencies, React will first run the cleanup function (if you provided it) with the old values, and then run your setup function with the new values. After your compo ...
- 
-* **optional** `dependencies`: The list of all reactive values referenced inside of the `setup` code. Reactive values include props, state, and all the variables and functions declared directly inside your component body. If your linter is [configured for React](/learn/editor-setup#linting), it will verify that every reactive value is correctly specified as a dependency. The list of dependencies  ...
+* `setup` (з англ. встановлюючий): Функція з логікою вашого Ефекту. Ваша функція `setup` може додатково повертати *cleanup* функцію (з англ. cleanup - прибирання). Після того, як ваш компонент буде додано в DOM, React виконає функцію `setup`. Потім після кожного наступного рендеру React буде перевіряти чи змінилось значення хоча б однієї із залежностей, і якщо значення хоча б однієї залежності зм ...
```
- тут і далі виправити небажане використання уюч аяч https://github.com/reactjs/uk.react.dev/issues/501
- тут і далі розібратися з написанням прикладок (React-компонент або компонент React, але не ~~React компонент~~)
- як тире використовувати саме тире, не дефіс

alinkedd commented on src/content/reference/react/useEffect.md:
```diff
@@ -40,45 +40,47 @@ function ChatRoom({ roomId }) {
+#### Параметри {/*parameters*/}
 
-* `setup`: The function with your Effect's logic. Your setup function may also optionally return a *cleanup* function. When your component is added to the DOM, React will run your setup function. After every re-render with changed dependencies, React will first run the cleanup function (if you provided it) with the old values, and then run your setup function with the new values. After your compo ...
- 
```
- параграфи і відступи мають збігатися з параграфами і відступами в українській версії, повернути порядок

alinkedd reviewed (changes requested):
Дякую за PR!

Це перша частина перевірки. Було перевірено загалом деякі слововживання чи правила.
Запити на зміни мають бути виправлені і перевірені у всій статті (та іншій статті про useEffectEvent) до того, як я почну вичитувати текст у наступних частинах перевірки, щоб зменшити кількість когнітивного навантаження від коректури та редактури того самого. Немає вимог, коли має бути виправлено, це буде просто послідовна асинхронна взаємодія перевірка-зміни-перевірка-зміни-перевірка тощо. Також через те, що я перевіряю одна і у мене майже немає часу, то довгі статті зливаються, на жаль, дуже довго.

Andrii256 committed:
Small fixes after PR

Andrii256 committed:
Виправлено прикладки

Andrii256 committed:
Змінено одні тупі імена на інші, ще тупіші 😀

Andrii256 committed:
Кілька дрібних змін для відповідності вимогам які були уточнені в іншому PR

"Події ефекту" (effectEvent) тепер з малої.

Заголовок "Довідка" змінено на "Опис"
